cpp/src/io/json/write_json.cu

```diff
@@ -142,6 +143,11 @@ struct escape_strings_fn {
       if (utf8_char > 0x0000'00FF) {
         // multi-byte char
         uint32_t codepoint = cudf::strings::detail::utf8_to_codepoint(utf8_char);
+        if (unescaped_utf8) {
+          // write original utf8 character if unescaping is enabled
+          write_char(utf8_char, d_buffer, bytes);
+          continue;
+        }
```

Comment:
Move this up before code point conversion.

Reply:
Done.